File: C++/760. Find Anagram Mappings.cpp
// Time:  O(n)
// Space: O(n)
class Solution {
public:
    vector<int> anagramMappings(vector<int>& A, vector<int>& B) {
        vector<int> ans;
        map<int, int> maple;
        int lenA = A.size();
        int lenB = B.size();
        for(int i=0;i<lenB;i++)
            maple[B[i]] = i;
        for(int i=0;i<lenA;i++)
            ans.push_back(maple[A[i]]);
        return ans;
    }
};

File: Python/760. Find Anagram Mappings.py
# Time:  O(n)
# Space: O(n)
class Solution:
    def anagramMappings(self, A, B):
        """
        :type A: List[int]
        :type B: List[int]
        :rtype: List[int]
        """
        ans = dict()
        for i in range(len(B)):
            ans[B[i]] = i
        return [ans[x] for x in A]